File: challenges/3-area-trapezio/go/teocalvo/challenge.go
package main

import (
	"fmt"
	"strconv"
)

func getNumber(msg string) float64 {
	var nString string
	for {
		fmt.Printf(msg)
		fmt.Scanf("%s", &nString)
		n, err := strconv.ParseFloat(nString, 64)
		if err != nil {
			fmt.Print("Entre com um número válido!\n")
			continue
		}
		return n
	}
}

func main() {
	baseMaior := getNumber("Entre com a base maior do trapézio: ")
	baseMenor := getNumber("Entre com a base menor do trapézio: ")
	altura := getNumber("Entre com a altura do trapézio: ")
	fmt.Printf("A área do trapézio é: %.2f\n", (baseMaior+baseMenor)*altura/2.)
}

File: challenges/3-area-trapezio/java/felipedelta0/Challenge.java
public class Challenge {
    public static double calculaAreaTrapezio(int baseMaior, int baseMenor, int altura) {
        return (baseMaior + baseMenor) * altura / 2.0;
    }

    public static void main(String[] args) {
        if (args == null || args.length < 3)
            return;

        int baseMaior = Integer.parseInt(args[0]);
        int baseMenor = Integer.parseInt(args[1]);
        int altura = Integer.parseInt(args[2]);

        System.out.println("A área do trapézio é: " + calculaAreaTrapezio(baseMaior, baseMenor, altura));
    }
}
